File: frontend/display_app/lib/models/category.dart
/// A product category / collection (e.g. "Ready-to-Wear", "Leather Goods").
class Category {
  const Category({
    required this.id,
    required this.name,
    this.tagline,
    this.imageUrl,
  });

  factory Category.fromJson(Map<String, dynamic> json) => Category(
        id: json['id'] as String,
        name: json['name'] as String,
        tagline: json['tagline'] as String?,
        imageUrl: json['imageUrl'] as String?,
      );

  final String id;
  final String name;
  final String? tagline;
  final String? imageUrl;

  @override
  bool operator ==(Object other) => other is Category && other.id == id;

  @override
  int get hashCode => id.hashCode;
}

File: frontend-rn/display_app/src/models/category.ts
/// A product category / collection. Ported from the Flutter `Category`.
export class Category {
  readonly id: string;
  readonly name: string;
  readonly tagline?: string;
  readonly imageUrl?: string;

  constructor(init: { id: string; name: string; tagline?: string; imageUrl?: string }) {
    this.id = init.id;
    this.name = init.name;
    this.tagline = init.tagline;
    this.imageUrl = init.imageUrl;
  }

  static fromJson(json: any): Category {
    return new Category({
      id: json.id as string,
      name: json.name as string,
      tagline: json.tagline as string | undefined,
      imageUrl: json.imageUrl as string | undefined,
    });
  }
}
